Update High CPU Tutorial (#29112)
dotnet/docs: docs/core/diagnostics/debug-highcpu.md

`https://localhost:5001/api/diagscenario/highcpu/60000`

> Now, rerun the [dotnet-counters](dotnet-counters.md) command. To monitor just the `cpu-usage`, add --counters `System.Runtime[cpu-usage]` as part of the command.

Reviewer: ```suggestion
Now, rerun the [dotnet-counters](dotnet-counters.md) command. To monitor just the `cpu-usage`, add `--counters System.Runtime[cpu-usage]` as part of the command.
```

```dotnetcli

> dotnet-counters monitor -p 22884 --refresh-interval 1

Reviewer: ```suggestion
dotnet-counters monitor -p 22884 --refresh-interval 1 --counters System.Runtime[cpu-usage]
```
Otherwise it's the same command as before. And as the text states "add" put it at the end -- not in the middle as the previous code was (left side of diff).

    Monitor Lock Contention Count / 1 sec                  0
    Number of Active Timers                                1
    Number of Assemblies Loaded                          140
    ThreadPool Completed Work Item Count / 1 sec           3
    ThreadPool Queue Length                                0
    ThreadPool Thread Count                                7
    Working Set (MB)                                      63
```

Throughout the duration of the request, the CPU usage will hover around the increased percentage.

> [!TIP]
> To visualize an even higher CPU usage, you can exercise this endpoint in multiple browser tabs simultaneously.

At this point, you can safely say the CPU is running higher than you expect. Identifying the effects of a problem is key to finding the cause. We will use the effect of high CPU consumption in addition to diagnostic tools to find the cause of the problem.

## Analyze High CPU with Profiler

When analyzing an app with high CPU usage, you need a diagnostics tool that can provide insights into what the code is doing. The usual choice is a profiler, and there are different profiler options to choose from.

### [Linux](#tab/linux)

> 

Reviewer 2: Should we explain why we suggest using the Linux perf tool and not dotnet-trace?  I'm guessing the reason is something like: Although Dotnet-Trace is cross-platform but since Event Pipe "CPU" traces only measure elapsed time and not actual CPU time.

Author: I had the same question. @josalem Why are we using the Linux perf tool and not dotnet-trace?

Reviewer 3: `dotnet-trace` is great for CPU investigations to a certain point. The limitations of safe-point bias and managed-only callstacks means that it can't give _exact_ information the same way a kernel-aware profiler like `perf` can.

If your performance investigation stops at managed code, generally `dotnet-trace` will be sufficient. If you are analyzing exact timing of your application and it involves native code, you may need something like `perf`. 

I think of it like two differently powered magnifying glasses. You can start with `dotnet-trace` and then look at a higher resolution using `perf` (or `ETW` on Windows).

    Status: Running

[System.Runtime]

>     % Time in GC since last GC (%)                         0

Reviewer: This does not match the text. Only cpu usage should be shown here.

    GC Heap Size (MB)                                      4
    Gen 0 GC Count / 60 sec                                0
    Gen 0 Size (B)                                         0
    Gen 1 GC Count / 60 sec                                0
    Gen 1 Size (B)                                         0
    Gen 2 GC Count / 60 sec                                0
    Gen 2 Size (B)                                         0
    LOH Size (B)                                           0
    Monitor Lock Contention Count / 1 sec                  0
    Number of Active Timers                                1
    Number of Assemblies Loaded                          140
    ThreadPool Completed Work Item Count / 1 sec           3
    ThreadPool Queue Length                                0
    ThreadPool Thread Count                                7
    Working Set (MB)                                      63
```

Throughout the duration of the request, the CPU usage will hover around 25% . Depending on the host machine, expect varying CPU usage.

> [!TIP]
> To visualize an even higher CPU usage, you can exercise this endpoint in multiple browser tabs simultaneously.

> At this point, you can safely say the CPU is running higher than you expect, meaning the app has a slow request. 

Reviewer: > meaning the app has a slow request.

There can be a myriad of reasons for higher cpu. Maybe write it as something like
```suggestion
At this point, you can safely say the CPU is running higher than you expect. With the knowledge of the app-code, and the requests made this means the app has a slow request. 
```
?
